includes/class-wc-checkout.php

```diff
@@ -697,13 +697,13 @@ protected function validate_posted_data( &$data, &$errors ) {
 				}
 
 				if ( in_array( 'email', $format, true ) && '' !== $data[ $key ] ) {
-					$data[ $key ] = sanitize_email( $data[ $key ] );
 
 					if ( $validate_fieldset && ! is_email( $data[ $key ] ) ) {
 						/* translators: %s: email address */
 						$errors->add( 'validation', sprintf( __( '%s is not a valid email address.', 'woocommerce' ), '<strong>' . esc_html( $field_label ) . '</strong>' ) );
 						continue;
 					}
+					$data[ $key ] = sanitize_email( $data[ $key ] );
```

Comment:
This will not sanitize the email if the email is invalid (because of the `continue`). 

I think it should be something like this:
```
$email_is_valid = is_email( $data[ $key ] );
$data[ $key ] = sanitize_email( $data[ $key ] );

if ( $validate_fieldset && ! $email_is_valid ) {
 	/* translators: %s: email address */
 	$errors->add( 'validation', sprintf( __( '%s is not a valid email address.', 'woocommerce' ), '<strong>' . esc_html( $field_label ) . '</strong>' ) );
 	continue;
}
```